Index: linyaps/libs/oci-cfg-generators/src/linglong/oci-cfg-generators/container_cfg_builder.cpp
===================================================================
--- linyaps.orig/libs/oci-cfg-generators/src/linglong/oci-cfg-generators/container_cfg_builder.cpp
+++ linyaps/libs/oci-cfg-generators/src/linglong/oci-cfg-generators/container_cfg_builder.cpp
@@ -368,7 +368,10 @@ ContainerCfgBuilder &ContainerCfgBuilder
         "/etc/localtime",
         "/etc/resolv.conf",
         "/etc/timezone",
-        "/etc/hosts"
+        "/etc/hosts",
+        "/usr/share/wallpapers",
+        "/usr/share/music",
+        "/usr/share/extensions-from-app-store"
     };
 
     hostStaticsMount = std::vector<Mount>{};
