Index: linyaps/libs/oci-cfg-generators/src/linglong/oci-cfg-generators/90_legacy.cpp
===================================================================
--- linyaps.orig/libs/oci-cfg-generators/src/linglong/oci-cfg-generators/90_legacy.cpp
+++ linyaps/libs/oci-cfg-generators/src/linglong/oci-cfg-generators/90_legacy.cpp
@@ -63,6 +63,8 @@ bool Legacy::generate(ocppi::runtime::co
         { "/usr/share/icons", "/usr/share/icons" },
         { "/usr/share/zoneinfo", "/usr/share/zoneinfo" },
         { "/etc/resolvconf", "/etc/resolvconf" },
+        { "/usr/share/wallpapers", "/usr/share/wallpapers" },
+        { "/usr/share/music", "/usr/share/music" },
     };
 
     for (const auto [source, destination] : roMountMap) {
